network/netidentity.go

```diff
@@ -94,12 +94,39 @@ type identityChallengeScheme interface {
 	VerifyResponse(h http.Header, c identityChallengeValue) (crypto.PublicKey, []byte, error)
 }
 
+type identityChallengeSigner interface {
```

Comment:
I think explaining when the legacy signer is used explicitly would be a good thing.

Reply:
I suggest renaming after merging https://github.com/algorand/go-algorand/pull/6035